File: parse_report.js
const fs = require('fs');

const data = JSON.parse(fs.readFileSync('eslint-report.json', 'utf8'));

const rules = {};
let total = 0;
for (const file of data) {
  for (const msg of file.messages || []) {
    const rule = msg.ruleId;
    if (rule) {
      rules[rule] = (rules[rule] || 0) + 1;
      total++;
    }
  }
}

console.log(`Total warnings/errors: ${total}`);
console.log("Warning/Error counts by rule:");
for (const [rule, count] of Object.entries(rules).sort((a, b) => b[1] - a[1])) {
  console.log(`  ${rule}: ${count}`);
}

File: parse_report.py
import json

with open("eslint-report.json", "r", encoding="utf-8") as f:
    data = json.load(f)

rules = {}
for file in data:
    for msg in file.get("messages", []):
        rule = msg.get("ruleId")
        if rule:
            rules[rule] = rules.get(rule, 0) + 1

print("Warning/Error counts by rule:")
for rule, count in sorted(rules.items(), key=lambda x: x[1], reverse=True):
    print(f"  {rule}: {count}")
